public/app/features/manage-dashboards/state/actions.ts

```diff
@@ -331,43 +332,42 @@ export function getFolderByUid(uid: string): Promise<{ uid: string; title: strin
 }
 
 export async function processV2DatasourceInput(
-  obj: PanelQueryKind['spec'] | QueryVariableKind['spec'] | AnnotationQueryKind['spec'],
+  spec: PanelQueryKind['spec'] | QueryVariableKind['spec'] | AnnotationQueryKind['spec'],
   inputs: Record<string, DataSourceInput> = {}
 ) {
-  const datasourceRef = obj?.datasource;
-  if (!datasourceRef && obj?.query) {
-    const dsType = obj.query.kind;
+  const datasourceRef = spec?.query?.datasource;
+  let dataSourceInput: DataSourceInput | undefined;
+  const dsType = spec.query?.group || 'undefined';
```

Comment:
@ivanortegaalba  - what is 'undefined' here? I don't think we ever used anything like a string undefined.